### PR TITLE
fix: 修复 attr 和 scale 不一样的情况

### PR DESCRIPTION
**Pull request by zengyue**
##### Checklist

- [ ] `npm test` passes
- [ ] tests and/or benchmarks are included
- [ ] commit message follows commit guidelines

### CODE REVIEW

File: packages/f2/test/components/geometry/attr.test.tsx [hunk 1]
```diff
@@ -263,25 +362,43 @@ describe('Geometry - Attr', () => {
...
           <Axis field="sales" />
-          <Point x="year" y="sales" size={12} color={['type', ['blue', 'red']]} />
+          <Point
+            ref={geometryRef}
+            x="year"
+            y="sales"
+            size={12}
+            color={{
+              field: 'sales',
+              type: 'category',
```
Comment (zengyue) [inline packages/f2/test/components/geometry/attr.test.tsx:390]: scale  是 linear, 用 category 映射的 case

File: packages/f2/test/components/geometry/attr.test.tsx [hunk 2]
```diff
@@ -164,93 +172,184 @@ describe('Geometry - Attr', () => {
...
+          <Axis field="year" />
+          <Axis field="sales" />
+          <Point
+            ref={geometryRef}
+            x="year"
+            y="sales"
+            size={12}
+            color={{
+              field: 'sales',
+              range: ['blue', 'red'],
```
Comment (zengyue) [inline packages/f2/test/components/geometry/attr.test.tsx:250]: 映射到颜色区间

File: packages/f2/test/components/geometry/attr.test.tsx [hunk 3]
```diff
@@ -164,93 +172,184 @@ describe('Geometry - Attr', () => {
...
       </Canvas>
     );
-    // @ts-ignore
-    const canvas = new type(props);
+    const canvas = new Canvas(props);
+    canvas.render();
 
+    expect(geometryRef.current.records.length).toBe(1);
+    // scales 的 scale 是 Linear， 有nice 处理
+    expect(geometryRef.current.records[0].children[0].color).toBe('rgb(32, 0, 223)');
```
Comment (zengyue) [inline packages/f2/test/components/geometry/attr.test.tsx:230]: 线性映射的断言

File: packages/f2/test/components/line/line.test.tsx [hunk 4]
```diff
@@ -467,60 +467,68 @@ describe('折线图', () => {
...
+      );
 
-          const canvas = new Canvas(props);
-          canvas.render();
-        });
+      const canvas = new Canvas(props);
+      canvas.render();
+
+      const line = lineRef.current;
+      expect(line.attrs.color.scale.values).toEqual(['金属', '农副产品', '能源']);
```
Comment (zengyue) [inline packages/f2/test/components/line/line.test.tsx:531]: values 的断言也加上了

**Review by El-Chiang — APPROVED** [review]
+1

**Pull request merged**